### ISSUE TITLE
Calling getTag in frontEndRule while using both file and consul-catalog backends causes file backend load to fail (WORKAROUND EXISTS)

### ISSUE THREAD

**Issue by rvaidya** [containous/traefik#3268]
### Do you want to request a *feature* or report a *bug*?

BUG

### What did you do?

If you look at the below config, I am using both the file and consul-catalog backends. Initially, I only had a single config file, `traefik.toml`.  After upgrading from 1.5.3, I started getting the following error when traefik started up, and the file backend would not load:
```
level=error msg="Error starting provider *file.Provider: template: /etc/traefik.d/traefik.toml:261: function \"getTag\" not defined"
```
It seems like my call to `getTag` in my consul-catalog `frontEndRule` was being picked up by the file backend.

I was able to work around this by using a separate file for the content of the file backend, `rules.toml`

### What did you expect to see?

Traefik would still work as before without needing changes.

### What did you see instead?

Started getting the following error when traefik started up, and the file backend would not load:
```
level=error msg="Error starting provider *file.Provider: template: /etc/traefik.d/traefik.toml:261: function \"getTag\" not defined"
```
### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.6.0
Codename:     tetedemoine
Go version:   go1.10.1
Built:        2018-04-30_09:28:44PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

<details>
<summary>config</summary>

```json
{
 "LifeCycle": {
  "RequestAcceptGraceTimeout": 0,
  "GraceTimeOut": 0
 },
 "GraceTimeOut": 0,
 "Debug": false,
 "CheckNewVersion": true,
 "SendAnonymousUsage": false,
 "AccessLogsFile": "",
 "AccessLog": null,
 "TraefikLogsFile": "",
 "TraefikLog": null,
 "Tracing": null,
 "LogLevel": "",
 "EntryPoints": {
  "http": {
   "Address": "xxxx",
   "TLS": null,
   "Redirect": {
    "entryPoint": "xxxx"
   },
   "Auth": null,
   "WhitelistSourceRange": null,
   "WhiteList": null,
   "Compress": true,
   "ProxyProtocol": null,
   "ForwardedHeaders": null
  },
  "https": {
   "Address": "xxxx",
   "TLS": {
    "MinVersion": "",
    "CipherSuites": null,
    "Certificates": null,
    "ClientCAFiles": null,
    "ClientCA": {
     "Files": null,
     "Optional": false
    }
   },
   "Redirect": null,
   "Auth": null,
   "WhitelistSourceRange": null,
   "WhiteList": null,
   "Compress": true,
   "ProxyProtocol": null,
   "ForwardedHeaders": null
  }
 },
 "Cluster": null,
 "Constraints": [],
 "ACME": {
  "Email": "xxxx",
  "Domains": [],
  "Storage": "xxxx",
  "StorageFile": "",
  "OnDemand": false,
  "OnHostRule": false,
  "CAServer": "xxxx",
  "EntryPoint": "xxxx",
  "DNSChallenge": null,
  "HTTPChallenge": null,
  "DNSProvider": "",
  "DelayDontCheckDNS": 0,
  "ACMELogging": false,
  "TLSConfig": null
 },
 "DefaultEntryPoints": [
  "http",
  "https"
 ],
 "ProvidersThrottleDuration": 2000000000,
 "MaxIdleConnsPerHost": 200,
 "IdleTimeout": 0,
 "InsecureSkipVerify": false,
 "RootCAs": null,
 "Retry": null,
 "HealthCheck": {
  "Interval": 30000000000
 },
 "RespondingTimeouts": null,
 "ForwardingTimeouts": null,
 "AllowMinWeightZero": false,
 "Web": {
  "Address": ":8080",
  "CertFile": "",
  "KeyFile": "",
  "ReadOnly": false,
  "Statistics": null,
  "Metrics": null,
  "Path": "",
  "Auth": null,
  "Debug": false
 },
 "Docker": null,
 "File": {
  "Watch": true,
  "Filename": "/etc/traefik.d/rules.toml",
  "Constraints": null,
  "Trace": false,
  "TemplateVersion": 0,
  "DebugLogGeneratedTemplate": false,
  "Directory": ""
 },
 "Marathon": null,
 "Consul": {
  "Watch": true,
  "Filename": "",
  "Constraints": [],
  "Trace": false,
  "TemplateVersion": 0,
  "DebugLogGeneratedTemplate": false,
  "Endpoint": "xxxx",
  "Prefix": "traefik",
  "TLS": null,
  "Username": "",
  "Password": ""
 },
 "ConsulCatalog": {
  "Watch": false,
  "Filename": "",
  "Constraints": [
   "tag==loadbalanced"
  ],
  "Trace": false,
  "TemplateVersion": 0,
  "DebugLogGeneratedTemplate": false,
  "Endpoint": "xxxx",
  "Domain": "xxxx",
  "ExposedByDefault": true,
  "Prefix": "traefik",
  "FrontEndRule": "Host:{{getTag \"traefik.subdomain\" .Attributes .ServiceName }}.{{.Domain}}",
  "TLS": null
 },
 "Etcd": null,
 "Zookeeper": null,
 "Boltdb": null,
 "Kubernetes": null,
 "Mesos": null,
 "Eureka": null,
 "ECS": null,
 "Rancher": null,
 "DynamoDB": null,
 "ServiceFabric": null,
 "Rest": null,
 "API": null,
 "Metrics": null,
 "Ping": null,
 "ConfigFile": "/etc/traefik.d/traefik.toml"
}
```

</details>

### If applicable, please paste the log output at DEBUG level (`--logLevel=DEBUG` switch)

```
(paste your output here)
```

**Comment by ldez** [comment]
Do you have a docker-compose to help us to reproduce?

**Comment by ldez** [comment]
I think it's a regression due to #2991

**Comment by rvaidya** [comment]
Sorry for not responding, I am not using docker, but I can provide my
config file.

**Comment by aantono** [comment]
@rvaidya Can you provide your traefik config file(s) and template(s) to help understand how the setup happens?

**Comment by traefiker** [comment]
Closed by #3347.
